pinot-core/src/main/java/org/apache/pinot/core/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -22,6 +22,7 @@
 import java.util.Iterator;
 import java.util.concurrent.ConcurrentHashMap;
 import javax.annotation.concurrent.ThreadSafe;
+import jdk.nashorn.internal.ir.debug.ObjectSizeCalculator;
```

Comment:
Is this supported in all JDK versions?

Reply:
Hi @fx19880617   I found ObjectSizeCalculator is not supported JDK11. 
https://mail.openjdk.java.net/pipermail/jdk8u-dev/2020-May/011770.html

I also looked into calculating the bytes used by metadata manually, but it's not optimal since it required extra storage.
Do you have any suggestions or recommendations, is there any similar metrics emitted by Pinot?

Reply:
I somehow feel you may need to add/minus the bytes when there is a primary add/delete/update.

Reply:
Also, please make sure the map size calculation is very fast because it happens on a per-record basis